Source/AwsCommonRuntimeKit/http/HTTPStreamCallbackCore.swift

```diff
@@ -66,9 +72,7 @@ private func onResponseHeaderBlockDone(stream: UnsafeMutablePointer<aws_http_str
                                        headerBlock: aws_http_header_block,
                                        userData: UnsafeMutableRawPointer!) -> Int32 {
     let httpStreamCbData = Unmanaged<HTTPStreamCallbackCore>.fromOpaque(userData).takeUnretainedValue()
-    let stream = httpStreamCbData.stream!
-
-    httpStreamCbData.requestOptions.onIncomingHeadersBlockDone(stream, HTTPHeaderBlock(rawValue: headerBlock))
+    httpStreamCbData.requestOptions.onIncomingHeadersBlockDone(HTTPHeaderBlock(rawValue: headerBlock))
```

Comment:
probably want to pass status in `onIncomingHeadersBlockDone` too. It's theoretically possible there are 0 headers, and so the onIncomingHeaders callback never fires

OR, while you're messing with the API, make it so there's only 1 `onIncomingHeaders` callback in Swift. Use the C on_incoming_headers callback to buffer up a list, then use the C on_incoming_headers_block_done callback to deliver those buffered up headers to Swift.

OR EVEN BETTER, still do the buffering thing, but have 3 separate callbacks: `onInterimResponse` (for 1xx INFORMATIONAL headers and status code), onResponse (for normal headers and status code), and onTrailer (for trailing headers (no status code)). This is how I wished I'd built the C APIs in retrospect. The scary thing with all different kinds of headers using the same callback is: if the user doesn't check the block type, their code will get real messed up if they ever get an informational 1xx interim response

That 3rd approach is *kind of* what we have In python ([python API](https://github.com/awslabs/aws-crt-python/blob/10d3900c2e88dfb491c9b93b75d47024685f9f66/awscrt/http.py#L177), [C code](https://github.com/awslabs/aws-crt-python/blob/10d3900c2e88dfb491c9b93b75d47024685f9f66/source/http_stream.c#L127-L137)). Except we never actually hooked up the `on_interim_response` and `on_trailer` callbacks, we only have the 1 `on_response` callback right now.